arch/arm/crc32_acle.c

```diff
@@ -29,45 +29,43 @@ Z_INTERNAL Z_TARGET_CRC uint32_t crc32_acle(uint32_t crc, const uint8_t *buf, si
             len--;
         }
 
-        if ((len >= sizeof(uint16_t)) && ((ptrdiff_t)buf & sizeof(uint16_t))) {
-            buf2 = (const uint16_t *) buf;
-            c = __crc32h(c, *buf2++);
+        if ((len >= sizeof(uint16_t)) && ((ptrdiff_t)buf & (sizeof(uint16_t) - 1))) {
+            buf2 = *((uint16_t*)buf);
+            c = __crc32h(c, buf2);
+            buf += sizeof(uint16_t);
             len -= sizeof(uint16_t);
-            buf4 = (const uint32_t *) buf2;
-        } else {
-            buf4 = (const uint32_t *) buf;
         }
 
-        if ((len >= sizeof(uint32_t)) && ((ptrdiff_t)buf & sizeof(uint32_t))) {
-            c = __crc32w(c, *buf4++);
+        if ((len >= sizeof(uint32_t)) && ((ptrdiff_t)buf & (sizeof(uint32_t) - 1))) {
```

Comment:
```c
if ((len >= sizeof(uint32_t)) && ((ptrdiff_t)buf & (sizeof(uint64_t) - 1))) {
```

This should resolve the alignment issue.

Reply:
Ahhh that makes sense, we still want to see if we're a modulus of 8 bytes, the prior adjustments in the loop are what guarantee we're aligned to the next steps up.  I'm testing this with a more reliable test (controlling the allocation in the benchmark input).  I should have the fix ready shortly if it proves correct for all alignments.

Reply:
Thanks for that, I was going crazy trying to figure out how this ever would have worked.  The trick is that even if there isn't a combination of aligned loads that get us to 8 byte alignment the order in which that's done means that it will at least be aligned to the next biggest size for the tail of the remainder peeling.